homeassistant/components/compit/climate.py

```diff
@@ -0,0 +1,282 @@
+            for item in self.available_presets.details
+            if item is not None
+        ]
+
+    @cached_property
+    def fan_modes(self) -> list[str] | None:
+        """Return the available fan modes."""
+        if self.available_fan_modes is None or self.available_fan_modes.details is None:
+            return []
+        return [
+            item.description
+            for item in self.available_fan_modes.details
+            if item is not None
+        ]
+
+    @property
+    def preset_mode(self) -> str | None:
+        """Return the current preset mode."""
+        preset_mode = self.coordinator.data[self.device_id].state.get_parameter_value(
+            PARAM_PRESET_MODE
+        )
+
+        if preset_mode is None:
+            return None
+        if self.available_presets is None or self.available_presets.details is None:
+            _LOGGER.debug("available_presets is None")
+            return None
+
+        val = next(
+            (
+                item
+                for item in self.available_presets.details
+                if item is not None and item.state == preset_mode.value
+            ),
+            None,
+        )
+        if val is None:
+            return None
+        return val.description
+
+    @property
+    def fan_mode(self) -> str | None:
+        """Return the current fan mode."""
+        fan_mode = self.coordinator.data[self.device_id].state.get_parameter_value(
+            PARAM_FAN_MODE
+        )
+        if fan_mode is None:
+            return None
+
+        if self.available_fan_modes is None or self.available_fan_modes.details is None:
+            _LOGGER.debug("available_fan_modes is None")
+            return None
+
+        val = next(
+            (
+                item
+                for item in self.available_fan_modes.details
+                if item is not None and item.state == fan_mode.value
+            ),
+            None,
+        )
+        if val is None:
+            return None
+        return val.description
+
+    @property
+    def hvac_mode(self) -> HVACMode | None:
+        """Return the current HVAC mode."""
+        hvac_mode = self.coordinator.data[self.device_id].state.get_parameter_value(
+            PARAM_HVAC_MODE
+        )
+        if hvac_mode:
+            if hvac_mode.value == HVAC_HEAT:
+                return HVACMode.HEAT
+            if hvac_mode.value == HVAC_OFF:
+                return HVACMode.OFF
+            if hvac_mode.value == HVAC_COOL:
+                return HVACMode.COOL
+        return None
+
+    async def async_set_temperature(self, **kwargs: Any) -> None:
+        """Set new target temperature."""
+        temp = kwargs.get(ATTR_TEMPERATURE)
+        if temp is None:
+            return
+        await self.async_call_api(PARAM_SET_TARGET_TEMPERATURE, temp)
+
+    async def async_set_hvac_mode(self, hvac_mode: HVACMode) -> None:
+        """Set new target HVAC mode."""
+        value = HVAC_HEAT
```

Comment:
if the mode is not one of the expected ones we should not fallback to `HEAT`. instead we should `raise ValueError`
also, if you want to, we could simplify this by using a map from HA HVACMode to compit HVAC modes. Check the `whirlpool` climate platform for an example